lib/Fhaculty/Graph/Walk.php

```diff
@@ -310,4 +310,25 @@ public function isValid()
 
         return true;
     }
+
+    /**
+     * @param Walk $walk
+     */
+    public function append(Walk $walk)
```

Comment:
The current `Walk` is implemented as an immutable object - this change adds the first method to make this mutable, which is something I'd like to avoid.

Instead, I'd like to change the method signature slightly to be more in line with the existing ones to something along the lines of:

```php
public static function factoryFromWalksAppended(Walk $first, Walk $second)
```

This should only affect your implementation slightly and yet have a meaningful effect on its semantics. Also, this makes it more obvious that the second `Walk` will be appended *behind* the first one.